lib/pages/chat_adaptive_scaffold/chat_adaptive_scaffold.dart

```diff
@@ -43,13 +55,19 @@ class _ChatAdaptiveScaffoldState extends State<ChatAdaptiveScaffold> {
                   end: breakpoint,
                 ): SlotLayout.from(
                   key: AppAdaptiveScaffold.breakpointMobileKey,
-                  builder: (_) => Stack(
-                    children: [
-                      body!,
+                  builder: (_) => Navigator(
+                    pages: [
+                      MaterialPage(
+                        child: body!,
+                      ),
                       if (showRightPanel)
-                        ChatSearch(
-                          roomId: widget.roomId,
-                          onBack: showRightPanelNotifier.toggle,
+                        MaterialPage(
+                          child: ChatSearch(
+                            roomId: widget.roomId,
+                            onBack: togglePage,
+                            jumpToEventId: jumpToEventId,
+                            isInStack: true,
+                          ),
```

Comment:
Can you explain it?

Reply:
push new screen without `push`, use `pages` array instead